### PR TITLE
[DOC] clarify how `fh` in forecasting methods is interpreted

### PR DESCRIPTION
**Pull request by ericjb**
Adds clearer documentation on use of `fh` in forecasters.

---

(old issue description, superseded)

Enhancement Issue #7182

Allows the predict method to pass an fh argument of type pd.Index. Internally this will coerce fh by 
setting fh = ForecastingHorizon(fh, is_relative=False). It provides convenience as in the following pseudo code:

fc = SomeForecaster(...)
fc.fit(y = train_data)
fitted = fc.predict(train_data.index)
forecasts = fc.predict(test_data.index)

Note that this makes the code tighter than if we required additional statements such as
fh = ForecastingHorizon(train_data.index, is_relative=False)

etc.

### CODE REVIEW

**Comment by yarnabrina** [comment]
@ericjb @fkiraly I am confused by this PR, I thought indices are already supported? Am I missing something super obvious?

```pycon
>>> from sktime.datasets import load_airline
>>> y = load_airline()
>>> from sktime.forecasting.naive import NaiveForecaster
>>> model = NaiveForecaster()
>>> model.fit(y)
NaiveForecaster()
>>> import pandas
>>> fh = pandas.Index([1, 2, 3])
>>> model.predict(fh=fh)
1961-01    432.0
1961-02    432.0
1961-03    432.0
Freq: M, Name: Number of airline passengers, dtype: float64
>>> type(fh)
<class 'pandas.core.indexes.base.Index'>
>>> isinstance(fh, pandas.Index)
True
```

**Comment by ericjb** [comment]
Hi @yarnabrina, that is really interesting, and something I never tried. According to the documentation of predict in forecasting/base/_base.py, the parameter fh is described as follows:

       Parameters
        ----------
        fh : int, list, np.array or ``ForecastingHorizon``, optional (default=None)
            The forecasting horizon encoding the time stamps to forecast at.

It does not say that pandas.Index is acceptable. But I tried it and it seems to work fine (based on a very limited sample). 

In this case, the PR can be ignored/closed/withdrawn. It would be nice for the documentation to be updated to reflect this capability.

Thanks

**Comment by fkiraly** [comment]
you could simply update the docs?

We also should add a test it in `test_fh`, it might not be properly tested there.

May I suggest changing this PR to that scope - docs and tests?

And, possibly, if tests fail, fixes?

**Comment by ericjb** [comment]
I don't think I am the best person to change the docs and write a test. I had no idea that this functionality is supported, I don't know where it is implemented, and I don't know its limitations, if any. 

Here is an idea for an additional enhancement. If the fh argument that is passed to predict is a pandas DataFrame or pandas Series, with a valid DatetimeIndex as index, that index could then be coerced to be the fh.

**Comment by fkiraly** [comment]
> I don't think I am the best person to change the docs and write a test.

Perhaps not, but you are certainly good enough by a significant margin? And you picked up on the issue that the docs are lacking.

> I had no idea that this functionality is supported, I don't know where it is implemented, and I don't know its limitations, if any.

I was also not aware! (at least not passively)
That's why we need to have docs and tests.

As a test, we could start by adding @yarnabrina's code to `test_fh`.

For the docs, we can just add `pd.Index` as a valid type to the docstrings containing `fh`.

File: sktime/forecasting/base/_base.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ class name: BaseForecaster
 Inspection methods:
     hyper-parameter inspection  - get_params()
     fitted parameter inspection - get_fitted_params()
-    current ForecastingHorizon  - fh
+    current ForecastngHorizon  - fh
```
Comment (yarnabrina) [inline sktime/forecasting/base/_base.py:26]:
```suggestion
    current ForecastingHorizon  - fh
```

The modification looks like a typo.

File: sktime/forecasting/base/_base.py [hunk 2]
```diff
@@ -418,11 +418,17 @@ def predict(self, fh=None, X=None):
 
         Parameters
         ----------
-        fh : int, list, np.array or ``ForecastingHorizon``, optional (default=None)
+        fh : int, list, np.array, pd.Index or ``ForecastingHorizon``,
```
Comment (yarnabrina) [inline sktime/forecasting/base/_base.py:421]: I'm okay with this change, but I'll note that I'm not 100% certain if it's really exhaustive types or not if that's the goal.

For example, recently `polars` is added, do you know if `.index` of that is allowed or not? May be @pranavvp16 knows.

File: sktime/forecasting/base/_base.py [hunk 3]
```diff
@@ -1885,6 +1919,9 @@ def _check_fh(self, fh, pred_int=False):
         else:
             # If fh is passed, coerce to ForecastingHorizon and validate (all cases)
 
+            if isinstance(fh, pd.Index):
```
Comment (yarnabrina) [inline sktime/forecasting/base/_base.py:1922]: Question: is it required? What happens if this block is absent?

Comment (ericjb) [reply sktime/forecasting/base/_base.py]: It is not required (I tested this). I left this code in by accident. This was supposed to be a "docstrings-only" PR. I removed this chunk and also fixed the typo. The goal is not to be exhaustive (too exhausting) but rather to improve the docstring incrementally to include a known omission. Re polars that is a great question but I have no idea (and no experience with polars).

File: sktime/forecasting/base/_base.py (hunk 2 above)
Comment (fkiraly) [reply sktime/forecasting/base/_base.py:421]: I think coercible iterables are fine.

**Pull request merged**